tests/func/test_datachain.py

```diff
@@ -25,34 +33,157 @@ def test_catalog_anon(tmp_dir, catalog, anon):
 
 def test_from_storage(cloud_test_catalog):
     ctc = cloud_test_catalog
-    dc = DataChain.from_storage(ctc.src_uri, catalog=ctc.catalog)
+    dc = DataChain.from_storage(ctc.src_uri, client_config=ctc.catalog.client_config)
```

Comment:
I agree about `Catalog`, but I'm not sure about session. If we go only with sessions, user will even for a simplest example need to construct `Session` object and will need to learn about another class / context manager etc. 
I think sessions are more for advanced usage like those ephemeral datasets where they are absolutely required, and we should allow users to do simple chains in the most easier way possible which in this case is providing simple config.
Also, I think we should remove `Catalog` from `Session` constructor as it could be confusing to the users.

BTW `client_config` also doesn't really sounds "right" to me. It would be more correct `cloud_config` IMO, but this is another discussion. 

WDYT?

Reply:
It would be great to know what configuration option we really need to expose to users (not unit tests). Off the top of my head, I can only think of anon.

In general, exposing multiple similar classes to uses seems like a big issue for users.

Reply:
Ok, I removed `client_config` and left only `anon` and `session`. User will need to create `Session` and add `client_config` to it if he want's to use different cloud credentials than set default in his environment (when using from CLI) which is ok I guess